lib/Runtime/Base/ScriptContext.cpp

```diff
@@ -1860,11 +1938,15 @@ namespace Js
                     Assert(segment != nullptr);
                     const uint32 offset = segment->getDestAddr();
                     const uint32 size = segment->getSourceSize();
-                    if (offset > maxSize || UInt32Math::Add(offset, size) >= maxSize)
+                    if (offset > maxSize || UInt32Math::Add(offset, size) > maxSize)
                     {
                         throw Wasm::WasmCompilationException(_u("Data segment #%u is out of bound"), iSeg);
                     }
-                    js_memcpy_s(buffer + offset, (uint32)maxSize - offset, segment->getData(), size);
+
+                    if (size > 0) 
```

Comment:
More a spec question, but I wonder if 0 size data segment should be considered compilation error?